pkg/crypto/signer.go

```diff
@@ -82,3 +91,37 @@ func (d *defaultSigner) Sign(data []byte) (signature []byte, err error) {
 	signature[64] = v
 	return signature, nil
 }
+
+// SignTx signs an ethereum transaction
+func (d *defaultSigner) SignTx(transaction *types.Transaction) (*types.Transaction, error) {
+	hash := (&types.HomesteadSigner{}).Hash(transaction).Bytes()
+	// isCompressedKey is false here so we get the expected v value (27 or 28)
+	signature, err := btcec.SignCompact(btcec.S256(), (*btcec.PrivateKey)(d.key), hash, false)
+	if err != nil {
+		return nil, err
+	}
+
+	// Convert to Ethereum signature format with 'recovery id' v at the end.
+	v := signature[0]
+	copy(signature, signature[1:])
+	// v value needs to be adjusted by 27 as transaction.WithSignature expects it to be 0 or 1
+	signature[64] = v - 27
+	return transaction.WithSignature(&types.HomesteadSigner{}, signature)
+}
+
+type EthAddress = [20]byte
```

Comment:
EthAddress is defined here, but in checkbook and other places common.Address is used. Since go-ethereum types is imported here, it should be ok to use common.Address from ethereum also?

Reply:
I changed it back to `common.Address`.